UnityXrefMaps/XrefMap.cs

```diff
@@ -47,15 +56,30 @@ public void FixHrefs(string apiUrl, bool testUrls = false)
                     continue;
                 }
 
+                reference.FixOverloadCommentId(references);
                 reference.FixHref(apiUrl);
                 fixedReferences.Add(reference);
+            }
+            references = fixedReferences.ToArray();
 
-                if (testUrls && !Utils.TestUriExists(reference.href).Result)
-                {
-                    Console.WriteLine("Warning: invalid URL " + reference.href + " for " + reference.uid + " uid");
+            if (testUrls) {
+                Console.WriteLine($"Testing URLs for {references.Length} references");
+
+                fixedReferences.Clear();
+                foreach (var reference in references) {
+                    Task<bool> testTask = Utils.TestUriExists(reference.href);
+                    testTask.Wait();
+                    if (testTask.Result) {
+                        fixedReferences.Add(reference);
+                    }
+                    else {
+                        Console.WriteLine("Warning: invalid URL " + reference.href + " for uid " + reference.uid);
```

Comment:
My bad here, I should use string interpolation:

```suggestion
                        Console.WriteLine($"Warning: invalid URL {reference.href} for uid {reference.uid}");
```